test/runtests.jl

```diff
@@ -32,4 +32,6 @@ using Test
 
 @time @testset "ODEs" begin include("ODEsTests/runtests.jl") end
 
+@time @testset "Adaptivity" begin include("AdaptivityTests/runtests.jl") end
```

Comment:
TO-THINK: I am not sure if putting all the adaptivity stuff in its own submodule it's the best way to go (from the point of view of dependencies and overall software architecture). At present, it fits well because it is brand new code, and thus all dependencies are pointing out to other gridap submodules above in the source code. But ... might it happen in the future as we develop AMR that we will be in need of dependencies pointing to Adaptivity from the other modules?

Reply:
Apologies. The previous comment should actually be attached to https://github.com/gridap/Gridap.jl/pull/838/files#diff-74b9bc0af4d59dbb92f6911b984a2b84e6d57014b0d341f1fd7bdffa09b4fae0R59

Reply:
Some comments that will come in handy to make an informed decision: 

As I see it, we have three choices: Either we leave `Adapted` as a high-level module (which depends on other modules but very few other modules depend on it) or we make it a low-level module. The third option would be to eliminate the module altogether. 

In the first case, we probably want to spread the low-level features around: 
 - `RefinementRule` and `CompositeQuadrature` make sense in `ReferenceFEs`.
 - `FineToCoarseField` makes sense in `Field`. 

This has a couple of problems: `FineToCoarseField` depends on `RefinementRule`. This would make `Fields` depend on `ReferenceFEs`. Also, `RefinementRule` depends on `Grid`. This would make `ReferenceFEs` depend on `Geometry`, which is bad since it's a circular dependence. This can be solved by moving `RefinementRule` into `Geometry`, but this would force `CompositeQuadrature` to be moved out as well (since the constructor depends on `RefinementRule`). Having a quadrature defined in `Geometry`does not make sense, in my opinion. 

The second option would be to keep the low-level features in `Adaptivity` and move the high-level features out: 
- `AdaptedDiscreteModel` and `AdaptedTriangulation` make sense in `Geometry`. 
- `AdaptivityGlue` could stay in `Adaptivity`, but would also make sense in `Geometry` since it is a glue that only makes sense when attached to a `Triangulation`. 

Again, this has the problem that `RefinementRule` depends on `Grid` and therefore `Geometry`...

Reply:
All in all, the main issue we have is that `RefinementRule` is something that is cell-wise (and therefore would make sense in `ReferenceFEs`) but it depends on a `Grid` in order to define the refinement. This creates a circular dependence in all cases. 
The obvious solution would be to get rid of this dependence, but I don't know if it's something we really want to do. It is probably possible, but not without some/a lot of code replication.